src/provider/cloud-element-templates/cmd/ChangeElementTemplateHandler.js

```diff
@@ -109,48 +108,13 @@ export default class ChangeElementTemplateHandler {
   }
 
   _updateZeebeModelerTemplateIcon(element, newTemplate) {
-    const bpmnFactory = this._bpmnFactory,
-          commandStack = this._commandStack;
+    const modeling = this._modeling;
 
     const icon = newTemplate && newTemplate.icon;
 
-    // remove existing icon when none new set
-    if (!newTemplate || !icon) {
-
-      const modelerTemplateIcon = findExtension(element, 'zeebe:ModelerTemplateIcon');
-
-      if (modelerTemplateIcon) {
-
-        const extensionElements = getBusinessObject(element).get('extensionElements');
 
-        commandStack.execute('element.updateModdleProperties', {
-          element,
-          moddleElement: extensionElements,
-          properties: {
-            values: without(extensionElements.get('values'), extension => extension === modelerTemplateIcon)
-          }
-        });
-      }
-
-      return;
-    }
-
-    const {
-      contents
-    } = icon;
-
-    // ensure extension elements
-    this._getOrCreateExtensionElements(element);
-
-    // create or update zeebe:modelerTemplateIcon
-    const newIcon = ensureExtension(element, 'zeebe:ModelerTemplateIcon', bpmnFactory);
-
-    return commandStack.execute('element.updateModdleProperties', {
-      element,
-      moddleElement: newIcon,
-      properties: {
-        body: contents
-      }
+    modeling.updateProperties(element, {
```

Comment:
much simpler even :-)

Reply:
Yeah, how easy it is to handle attributes instead of complex extension elements 🙂